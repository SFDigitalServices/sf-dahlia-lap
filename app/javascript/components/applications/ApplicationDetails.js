import React from 'react'
import ApplicationDetailsContentCard from './ApplicationDetailsContentCard'
import ApplicationDetailsContentTable from './ApplicationDetailsContentTable'

const ApplicationDetails = ({ application, fields, file_base_url }) => {

  const editLink = () => {
    if (!application.Is_Lottery_Complete && application.Application_Submission_Type == 'Paper') {
      return (
        <p><a href={`/applications/${application.Id}/edit`}>Edit Application</a></p>
      )
    }
  }

  let proofList = _.map(application.proof_files, (file) => {
    return <li key={file.Id}><a target='_blank' href={`${file_base_url}/servlet/servlet.FileDownload?file=${file.Id}`}>{file.Name}</a></li>
  })

  return (
    <div>
      {editLink()}
      <ApplicationDetailsContentCard
        dataCollection={application}
        title='Application Data'
        fields={[
          'Lottery_Number',
          'Application_Submission_Type',
          'Application_Submitted_Date',
          'Application_Language',
          'Total_Household_Size',
          'Total_Monthly_Rent',
          'Status',
          'Referral_Source',
          'CreatedBy.Name'
        ]}
        labelMapper={fields}
      />
      <ApplicationDetailsContentCard
        dataCollection={application.Applicant}
        title='Primary Applicant'
        fields={[
          'First_Name',
          'Last_Name',
          'Date_of_Birth',
          'Phone',
          'Second_Phone',
          'Email',
          'Residence_Address',
          'Mailing_Address',
        ]}
      />
      <ApplicationDetailsContentCard
        dataCollection={application.Alternate_Contact}
        title='Alternate Contact'
        fields={[
          'First_Name',
          'Last_Name',
          'Phone',
          'Agency_Name',
          'Email',
          'Alternate_Contact_Type',
          'Alternate_Contact_Type_Other'
        ]}
      />
      {(() => {
        if (application.household_members) {
          return (
            <ApplicationDetailsContentTable
              data={application}
              title='Household Members'
              table='household_members'
              fields={[
                'Name',
                'Relationship_to_Applicant',
                'Date_of_Birth',
                'Street',
                'City',
                'State',
                'Zip_Code'
              ]}
            />
          )
        }
      })()}
      <ApplicationDetailsContentCard
        dataCollection={application}
        title='Reserved and Priority Qualifying Information'
        fields={[
          'Has_Military_Service',
          'Has_DevelopmentalDisability',
          'Has_ADA_Priorities_Selected',
          'Answered_Community_Screening'
        ]}
        labelMapper={fields}
      />
      {(() => {
        if (application.preferences) {
          return (
            <ApplicationDetailsContentTable
              data={application}
              title='Application Preferences'
              table='preferences'
              fields={[
                'Receives_Preference',
                'Preference_Name',
                'Lottery_Status',
                'Person_who_claimed_Name',
                'Type_of_proof',
                'Preference_Lottery_Rank',
                'Opt_Out'
              ]}
            />
          )
        }
      })()}
      <ApplicationDetailsContentCard
        dataCollection={application}
        title='Declared Household Income'
        fields={[
          'Annual_Income',
          'Monthly_Income',
          'Housing_Voucher_or_Subsidy'
        ]}
      />
      {(() => {
        if (application.flagged_applications) {
          return (
            <ApplicationDetailsContentTable
              data={application}
              title='Flagged Applications'
              table='flagged_applications'
              fields={[
                'Flagged_Record_Set.Rule_Name',
                'Flagged_Record_Set.Total_Number_of_Pending_Review',
                'View Record Set'
              ]}
            />
          )
        }
      })()}
      <div className="content-card">
        <h4 className="content-card_title t-serif">Attachments</h4>
        <ul>{proofList}</ul>
      </div>
    </div>
  )
}

export default ApplicationDetails
