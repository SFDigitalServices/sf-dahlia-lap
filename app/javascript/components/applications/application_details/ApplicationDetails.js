import React from 'react'

import { map, isEmpty } from 'lodash'

import appPaths from 'utils/appPaths'

import ApplicationDetailsContentCard from './ApplicationDetailsContentCard'
import ApplicationDetailsContentTable from './ApplicationDetailsContentTable'
import {
  flaggedApplicationsFields,
  declareHouseholdIncome,
  applicationDataFields,
  primaryApplicantFields,
  alternateContactFields,
  householdMembersFields,
  applicationPreferencesFields,
  reservedAndPriorityFields,
  applicationEligibilityFields
} from './fields'

const ProofListItems = ({ fileBaseUrl, files }) =>
  map(files, (file) => {
    return (
      <li key={file.id}>
        <a target='_blank' rel='noreferrer' href={appPaths.toAttachmentDownload(fileBaseUrl, file)}>
          {file.document_type}
        </a>
      </li>
    )
  })

const ApplicationDetails = ({ application, fields, fileBaseUrl }) => {
  const ApplicationCard = (props) => (
    <ApplicationDetailsContentCard dataCollection={application} {...props} />
  )
  const Table = (props) => <ApplicationDetailsContentTable data={application} {...props} />

  return (
    <div className='application-details'>
      <ApplicationCard
        title='Application Data'
        fields={applicationDataFields}
        labelMapper={fields}
      />
      <ApplicationDetailsContentCard
        dataCollection={application.applicant}
        title='Primary Applicant'
        fields={primaryApplicantFields}
      />
      {application.listing.is_sale && (
        <ApplicationDetailsContentCard
          title='Eligibility'
          dataCollection={application}
          fields={applicationEligibilityFields}
          splitOn={3}
          labelMapper={fields}
        />
      )}
      <ApplicationDetailsContentCard
        dataCollection={application.alternate_contact}
        title='Alternate Contact'
        fields={alternateContactFields}
      />
      {!isEmpty(application.household_members) && (
        <Table
          title='Household Members'
          table='household_members'
          fields={householdMembersFields}
        />
      )}
      <ApplicationCard
        title='Reserved and Priority Qualifying Information'
        fields={reservedAndPriorityFields}
        labelMapper={fields}
      />
      {!isEmpty(application.preferences) && (
        <Table
          title='Application Preferences'
          table='preferences'
          fields={applicationPreferencesFields}
        />
      )}
      <ApplicationCard title='Declared Household Income' fields={declareHouseholdIncome} />
      {!isEmpty(application.flagged_applications) && (
        <Table
          title='Flagged Applications'
          table='flagged_applications'
          fields={flaggedApplicationsFields}
        />
      )}
      <div className='content-card'>
        <h4 className='content-card_title t-serif'>Attachments</h4>
        <ul>
          <ProofListItems files={application.proof_files} fileBaseUrl={fileBaseUrl} />
        </ul>
      </div>
    </div>
  )
}

export default ApplicationDetails
