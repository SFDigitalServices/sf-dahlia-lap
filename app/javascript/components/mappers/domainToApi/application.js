import { toPairs, filter, map, join, compact } from 'lodash'
import { createFieldMapper } from '~/utils/objectUtils'
const checkListToString = (list) => {
  return join(compact(list), ';')
}

const adaPrioritiesMapValues = {
  'vision_impairments': 'Vision impairments',
  'mobility_impairments': 'Mobility impairments',
  'hearing_impairments': 'Hearing impairments'
}

const mapAdaPrioritiesMap = (list) => {
  const selectedPriorities = filter(toPairs(list), ([a, b]) => b)
  return map(selectedPriorities, ([a, b]) => adaPrioritiesMapValues[a])
}

export const applicationFieldMapper = {
  id: 'id',
  primary_applicant_contact: 'primaryApplicantContact',
  application_language: 'applicationLanguage',
  has_military_service: 'hasMilitaryService',
  has_developmental_disability: 'hasDevelopmentalDisability',
  answered_community_screening: 'answeredCommunityScreening',
  housing_voucher_or_subsidy: 'householdVouchersSubsidies',
  terms_acknowledged: 'agreeToTerms',
  number_of_dependents: 'numberOfDependents',
  number_of_seniors: 'numberOfSeniors',
  number_of_minors: 'numberOfMinors',
  reserved_senior: 'hasSenior',
  application_submission_type: 'applicationSubmissionType',
  processing_status: 'processingStatus',
  household_assets: 'householdAssets',
  annual_income: 'annualIncome',
  monthly_income: 'monthlyIncome',
  confirmed_household_annual_income: 'confirmedHouseholdAnnualIncome',
  hh_total_income_with_assets_annual: 'HHTotalIncomeWithAssets',
  total_monthly_rent: 'totalMonthlyRent',
  adaPrioritiesSelected: (source) => checkListToString(mapAdaPrioritiesMap(source.has_ada_priorities_selected)),
  is_first_time_homebuyer: 'isFirstTimeHomebuyer',
  has_completed_homebuyer_education: 'hasCompletedHomebuyerEducation',
  has_loan_preapproval: 'hasLoanPreapproval',
  lending_agent: 'lendingAgent'
}

export const mapApplication = createFieldMapper(applicationFieldMapper)
