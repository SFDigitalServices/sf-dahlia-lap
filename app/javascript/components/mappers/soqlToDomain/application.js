import { split, snakeCase, fromPairs } from 'lodash'
import { mapApplicationMember } from './application_member'
import { mapApplicationPreference } from './application_preference'
import { mapAttachment } from './attachment'
import { mapDemographics } from './demographics'
import { mapFlaggedApplication } from './flagged_application'
import { mapListing } from './listing'
import { mapUser } from './user'
import { mapShape, mapList } from '../utils'
import formUtils from '~/utils/formUtils'

const parseList = text => split(text, ';')
const toChecklist = list => fromPairs(list.map(i => [snakeCase(i), true]))

export const mapApplication = (a) => {
  return {
    applicant: mapShape(mapApplicationMember, a.Applicant),
    demographics: mapShape(mapDemographics, a.Applicant),
    primary_applicant_contact: a.Primary_Applicant,
    alternate_contact: mapShape(mapApplicationMember, a.Alternate_Contact),
    listing: mapShape(mapListing, a.Listing),
    preferences: mapList(mapApplicationPreference, a.preferences),
    proof_files: mapList(mapAttachment, a.proof_files),
    household_members: mapList(mapApplicationMember, a.household_members),
    flagged_applications: mapList(mapFlaggedApplication, a.flagged_applications),
    lease: a.lease,
    rental_assistances: a.rental_assistances,
    createdby: mapShape(mapUser, a.CreatedBy),
    id: a.Id,
    name: a.Name,
    status: a.Status,
    total_household_size: a.Total_Household_Size,
    application_submission_type: a.Application_Submission_Type,
    application_submitted_date: a.Application_Submitted_Date,
    annual_income: formUtils.formatPrice(a.Annual_Income),
    confirmed_household_annual_income: formUtils.formatPrice(a.Confirmed_Household_Annual_Income),
    hh_total_income_with_assets_annual: formUtils.formatPrice(a.HH_Total_Income_with_Assets_Annual),
    household_assets: formUtils.formatPrice(a.Household_Assets),
    monthly_income: a.Monthly_Income,
    housing_voucher_or_subsidy: a.Housing_Voucher_or_Subsidy,
    referral_source: a.Referral_Source,
    application_language: a.Application_Language,
    lottery_number_manual: a.Lottery_Number_Manual,
    lottery_number: a.Lottery_Number,
    total_monthly_rent: formUtils.formatPrice(a.Total_Monthly_Rent),
    general_lottery: a.General_Lottery,
    general_lottery_rank: a.General_Lottery_Rank,
    answered_community_screening: a.Answered_Community_Screening,
    has_military_service: a.Has_Military_Service,
    has_developmental_disability: a.Has_DevelopmentalDisability,
    has_ada_priorities_selected: toChecklist(parseList(a.Has_ADA_Priorities_Selected)),
    terms_acknowledged: a.Terms_Acknowledged,
    number_of_dependents: a.Number_of_Dependents,
    number_of_seniors: a.Number_of_Seniors,
    number_of_minors: a.Number_of_Minors,
    processing_status: a.Processing_Status,
    status_last_updated: a.Status_Last_Updated,
    reserved_senior: a.Reserved_Senior,
    is_first_time_homebuyer: a.Is_First_Time_Homebuyer,
    has_completed_homebuyer_education: a.Has_Completed_Homebuyer_Education,
    has_loan_preapproval: a.Has_Loan_Pre_approval,
    lending_agent: a.Lending_Agent
  }
}
