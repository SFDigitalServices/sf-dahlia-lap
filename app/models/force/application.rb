# frozen_string_literal: true

# TODO: Re-enable Rubocop's line length check on this file once
# we have settled on a format for the field mapping items
# rubocop:disable Metrics/LineLength
module Force
  # Represent an application object. Provide mapping between
  # Salesforce object field names, Salesforce custom API field names,
  # and LAP domain field names for applications.
  class Application < Force::ObjectBase
    # TODO: Once we add more models and more fields, consider moving the
    # field mappings into YML files or other places.
    FIELD_NAME_MAPPINGS = [
      { custom_api: 'adaPrioritiesSelected', domain: 'has_ada_priorities_selected', salesforce: 'Has_ADA_Priorities_Selected' },
      { custom_api: 'agreeToTerms', domain: 'terms_acknowledged', salesforce: 'Terms_Acknowledged' },
      { custom_api: 'annualIncome', domain: 'annual_income', salesforce: 'Annual_Income' },
      { custom_api: 'answeredCommunityScreening', domain: 'answered_community_screening', salesforce: 'Answered_Community_Screening' },
      { custom_api: 'applicationLanguage', domain: 'application_language', salesforce: 'Application_Language' },
      { custom_api: 'applicationSubmissionType', domain: 'application_submission_type', salesforce: 'Application_Submission_Type' },
      { custom_api: 'applicationSubmittedDate', domain: 'application_submitted_date', salesforce: 'Application_Submitted_Date' },
      { custom_api: 'appRTType', domain: '', salesforce: '' },
      { custom_api: '', domain: 'createdby', salesforce: 'CreatedBy' },
      { custom_api: 'didApplicantUseHousingCounselingAgency', domain: '', salesforce: 'Applicant_used_housing_counseling_agency' },
      { custom_api: 'externalSessionId', domain: '', salesforce: 'Third_Party_External_ID' },
      { custom_api: 'finalHouseholdIncome', domain: '', salesforce: 'Final_Household_Income' },
      { custom_api: 'formMetadata', domain: '', salesforce: '' },
      { custom_api: '', domain: 'general_lottery', salesforce: 'General_Lottery' },
      { custom_api: '', domain: 'general_lottery_rank', salesforce: 'General_Lottery_Rank' },
      { custom_api: 'hasDevelopmentalDisability', domain: 'has_developmental_disability', salesforce: 'Has_DevelopmentalDisability' },
      { custom_api: 'hasMilitaryService', domain: 'has_military_service', salesforce: 'Has_Military_Service' },
      { custom_api: 'hasPublicHousing', domain: '', salesforce: '' },
      { custom_api: 'hasSenior', domain: 'reserved_senior', salesforce: 'Reserved_Senior' },
      { custom_api: 'HHTotalIncomeWithAssets', domain: 'hh_total_income_with_assets_annual', salesforce: 'HH_Total_Income_with_Assets_Annual' },
      { custom_api: 'householdAssets', domain: 'household_assets', salesforce: 'Household_Assets' },
      { custom_api: 'householdVouchersSubsidies', domain: 'housing_voucher_or_subsidy', salesforce: 'Housing_Voucher_or_Subsidy' },
      { custom_api: 'housingCounselingAgency', domain: '', salesforce: 'Which_One' },
      { custom_api: 'id', domain: 'id', salesforce: 'Id' },
      { custom_api: 'interviewScheduledDate', domain: '', salesforce: 'Interview_Scheduled_Date' },
      { custom_api: '', domain: 'is_lottery_complete', salesforce: 'Is_Lottery_Complete' },
      { custom_api: 'listingID', domain: '', salesforce: 'Listing' },
      { custom_api: 'lotteryNumber', domain: 'lottery_number', salesforce: 'Lottery_Number' },
      { custom_api: 'lotteryNumberManual', domain: 'lottery_number_manual', salesforce: 'Lottery_Number_Manual' },
      { custom_api: 'monthlyIncome', domain: 'monthly_income', salesforce: 'Monthly_Income' },
      { custom_api: 'name', domain: 'name', salesforce: 'Name' },
      { custom_api: 'numberOfDependents', domain: 'number_of_dependents', salesforce: 'Number_of_Dependents' },
      { custom_api: 'numberOfSeniors', domain: 'number_of_seniors', salesforce: 'Number_of_Seniors' },
      { custom_api: 'numberOfMinors', domain: 'number_of_minors', salesforce: 'Number_of_Minors' },
      { custom_api: 'otherHousingCounselingAgency', domain: '', salesforce: 'Other' },
      { custom_api: 'primaryApplicantContact', domain: 'primary_applicant_contact', salesforce: 'Primary_Applicant' },
      { custom_api: 'processingStatus', domain: 'processing_status', salesforce: 'Processing_Status' },
      { custom_api: 'referral', domain: 'referral_source', salesforce: 'Referral_Source' },
      { custom_api: 'snapshotId', domain: '', salesforce: 'Snapshot_ID' },
      { custom_api: 'status', domain: 'status', salesforce: 'Status' },
      { custom_api: '', domain: 'status_last_updated', salesforce: 'Status_Last_Updated' },
      { custom_api: 'subStatus', domain: 'sub_status', salesforce: 'Sub_Status' },
      { custom_api: '', domain: 'total_household_size', salesforce: 'Total_Household_Size' },
      { custom_api: 'totalMonthlyRent', domain: 'total_monthly_rent', salesforce: 'Total_Monthly_Rent' },
    ].freeze

    def to_domain
      domain_fields = super

      # Special field conversion cases for applications
      existing_fields = @fields[:salesforce].presence || @fields[:custom_api].presence
      if existing_fields.present?
        # ADA priorities
        ada_value = existing_fields['Has_ADA_Priorities_Selected'] || existing_fields['adaPrioritiesSelected']
        if ada_value
          ada_arr = ada_value.split(';')
          ada_hash = ada_arr.map { |x| [x.parameterize(separator: '_'), true] }.to_h
          domain_fields['has_ada_priorities_selected'] = ada_hash
        end

        # Created by
        domain_fields['createdby'] = { name: existing_fields['CreatedBy']['Name'] } if existing_fields['CreatedBy']
      end

      domain_fields
    end

    # TODO: There is code somewhere in this app that causes the Salesforce field
    # names we use in SOQL queries to have the "__c" suffix automatically added
    # onto them. That forces the field mappings in these Force objects to use
    # Salesforce field names without that suffix. That in turn causes issues
    # when trying to use the field mappings to translate to Salesforce fields
    # for REST API calls. Find and remove the code that strips out the "__c"
    # suffix, so that the Force objects' field mappings can contain true
    # Salesforce field names. Then remove this function as it won't be needed.
    def to_salesforce
      salesforce_fields = super
      add_salesforce_suffix(salesforce_fields)
    end
  end
end
# rubocop:enable Metrics/LineLength
