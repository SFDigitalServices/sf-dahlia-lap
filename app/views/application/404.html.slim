h1 Record not found
