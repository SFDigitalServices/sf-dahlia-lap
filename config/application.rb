# frozen_string_literal: true

require_relative 'boot'

require 'rails/all'

# Require the gems listed in Gemfile, including any gems
# you've limited to :test, :development, or :production.
Bundler.require(*Rails.groups)

module SfDahliaLap
  # Top level class for LAP application.
  class Application < Rails::Application
    # Initialize configuration defaults for originally generated Rails version.
    config.load_defaults 5.1
    # update defaults when we are done with config/initializers/new_framework_defaults_7_0.rb
    # config.load_defaults 7.0

    # TODO: remove this once we are on Rails 7, only needed as we incrementally upgrade
    config.autoloader = :zeitwerk

    # Disables the deprecated #to_s override in some Ruby core classes
    # See https://guides.rubyonrails.org/configuring.html#config-active-support-disable-to-s-conversion for more information.
    config.active_support.disable_to_s_conversion = true

    # Change the format of the cache entry to 7.0 after deploying the 7.0 upgrade
    config.active_support.cache_format_version = 7.0

    # Rails 7 can protect from open redirect attacks in `redirect_back_or_to` and `redirect_to`.
    # This is not compatible with our authentication process so we disable it
    # config.action_controller.raise_on_open_redirects = false # TODO do we need this?

    # Settings in config/environments/* take precedence over those specified here.
    # Application configuration should go into files in config/initializers
    # -- all .rb files in that directory are automatically loaded.

    config.assets.quiet = true
    config.action_view.logger = nil
  end
end
