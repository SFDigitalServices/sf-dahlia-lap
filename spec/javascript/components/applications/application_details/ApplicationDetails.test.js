import React from 'react'
import ApplicationDetails from 'components/applications/application_details/ApplicationDetails'
import application from '../../../fixtures/application'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'

describe('ApplicationDetails', () => {
  describe('should render', () => {
    test('without error and as expected', () => {
      const fileBaseUrl = 'http://www.someurl.com'

      const wrapper = renderer.create(
        <ApplicationDetails application={application} file_base_url={fileBaseUrl} />
      )
      expect(wrapper.toJSON()).toMatchSnapshot()
    })

    test('file links with both url types', () => {
      const fileBaseUrl = 'http://www.someurl.com'

      const wrapper = mount(
        <ApplicationDetails application={application} file_base_url={fileBaseUrl} />
      )
      expect(wrapper.find('a').first().getDOMNode().getAttribute('href')).toContain(
        'servlet/servlet.FileDownload'
      )
      expect(wrapper.find('a').last().getDOMNode().getAttribute('href')).toContain(
        'sfc/servlet.shepherd/version/download'
      )
    })
  })
})
