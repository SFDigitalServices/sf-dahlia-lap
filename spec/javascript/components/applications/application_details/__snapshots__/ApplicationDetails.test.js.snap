// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationDetails should render without error and as expected 1`] = `
<div
  className="application-details"
>
  <div
    className="content-card padding-bottom-none margin-bottom--half bg-trans"
  >
    <h4
      className="content-card_title t-serif"
    >
      Application Data
    </h4>
    <ul
      className="content-grid"
    >
      <li
        className="content-item"
      >
        <div
          className="content-card"
        >
          <div
            className="margin-bottom--half"
          >
            <h4
              className="t-sans t-small t-bold no-margin"
            >
              Lottery Number
            </h4>
            <p>
              00191270
            </p>
          </div>
          <div
            className="margin-bottom--half"
          >
            <h4
              className="t-sans t-small t-bold no-margin"
            >
              Application Submission Type
            </h4>
            <p>
              Electronic
            </p>
          </div>
          <div
            className="margin-bottom--half"
          >
            <h4
              className="t-sans t-small t-bold no-margin"
            >
              Application Submitted Date
            </h4>
            <p>
              06/19/2018
            </p>
          </div>
          <div
            className="margin-bottom--half"
          >
            <h4
              className="t-sans t-small t-bold no-margin"
            >
              Application Language
            </h4>
            <p>
              English
            </p>
          </div>
        </div>
      </li>
      <li
        className="content-item"
      >
        <div
          className="content-card"
        >
          <div
            className="margin-bottom--half"
          >
            <h4
              className="t-sans t-small t-bold no-margin"
            >
              Total Household Size
            </h4>
            <p>
              2
            </p>
          </div>
          <div
            className="margin-bottom--half"
          >
            <h4
              className="t-sans t-small t-bold no-margin"
            >
              Total Monthly Rent
            </h4>
            <p>
              None
            </p>
          </div>
          <div
            className="margin-bottom--half"
          >
            <h4
              className="t-sans t-small t-bold no-margin"
            >
              Status
            </h4>
            <p>
              Submitted
            </p>
          </div>
          <div
            className="margin-bottom--half"
          >
            <h4
              className="t-sans t-small t-bold no-margin"
            >
              Referral Source
            </h4>
            <p>
              None
            </p>
          </div>
          <div
            className="margin-bottom--half"
          >
            <h4
              className="t-sans t-small t-bold no-margin"
            >
              Created By
            </h4>
            <p>
              Prod Vertiba
            </p>
          </div>
        </div>
      </li>
    </ul>
  </div>
  <div
    className="content-card padding-bottom-none margin-bottom--half bg-trans"
  >
    <h4
      className="content-card_title t-serif"
    >
      Primary Applicant
    </h4>
    <ul
      className="content-grid"
    >
      <li
        className="content-item"
      >
        <div
          className="content-card"
        >
          <div
            className="margin-bottom--half"
          >
            <h4
              className="t-sans t-small t-bold no-margin"
            >
              First Name
            </h4>
            <p>
              karen
            </p>
          </div>
          <div
            className="margin-bottom--half"
          >
            <h4
              className="t-sans t-small t-bold no-margin"
            >
              Middle Name
            </h4>
            <p>
              elizabeth
            </p>
          </div>
          <div
            className="margin-bottom--half"
          >
            <h4
              className="t-sans t-small t-bold no-margin"
            >
              Last Name
            </h4>
            <p>
              jones
            </p>
          </div>
          <div
            className="margin-bottom--half"
          >
            <h4
              className="t-sans t-small t-bold no-margin"
            >
              Date of Birth
            </h4>
            <p>
              01/01/1950
            </p>
          </div>
          <div
            className="margin-bottom--half"
          >
            <h4
              className="t-sans t-small t-bold no-margin"
            >
              Email
            </h4>
            <p>
              eee@eeee.com
            </p>
          </div>
        </div>
      </li>
      <li
        className="content-item"
      >
        <div
          className="content-card"
        >
          <div
            className="margin-bottom--half"
          >
            <h4
              className="t-sans t-small t-bold no-margin"
            >
              Phone
            </h4>
            <p>
              9548883321
            </p>
          </div>
          <div
            className="margin-bottom--half"
          >
            <h4
              className="t-sans t-small t-bold no-margin"
            >
              Phone Type
            </h4>
            <p>
              Cell
            </p>
          </div>
          <div
            className="margin-bottom--half"
          >
            <h4
              className="t-sans t-small t-bold no-margin"
            >
              Second Phone
            </h4>
            <p>
              None
            </p>
          </div>
          <div
            className="margin-bottom--half"
          >
            <h4
              className="t-sans t-small t-bold no-margin"
            >
              Second Phone Type
            </h4>
            <p>
              None
            </p>
          </div>
          <div
            className="margin-bottom--half"
          >
            <h4
              className="t-sans t-small t-bold no-margin"
            >
              Residence Address
            </h4>
            <p>
              123 MAIN ST, SAN FRANCISCO, CA, 94105-1804
            </p>
          </div>
          <div
            className="margin-bottom--half"
          >
            <h4
              className="t-sans t-small t-bold no-margin"
            >
              Mailing Address
            </h4>
            <p>
              123 Mailing St, SF Mailing, CA, 94105-1804
            </p>
          </div>
        </div>
      </li>
    </ul>
  </div>
  <div
    className="content-card padding-bottom-none margin-bottom--half bg-trans"
  >
    <h4
      className="content-card_title t-serif"
    >
      Alternate Contact
    </h4>
    <ul
      className="content-grid"
    >
      <li
        className="content-item"
      >
        <div
          className="content-card"
        >
          <div
            className="margin-bottom--half"
          >
            <h4
              className="t-sans t-small t-bold no-margin"
            >
              First Name
            </h4>
            <p>
              Federic
            </p>
          </div>
          <div
            className="margin-bottom--half"
          >
            <h4
              className="t-sans t-small t-bold no-margin"
            >
              Middle Name
            </h4>
            <p>
              Daaaa
            </p>
          </div>
          <div
            className="margin-bottom--half"
          >
            <h4
              className="t-sans t-small t-bold no-margin"
            >
              Last Name
            </h4>
            <p>
              dayan
            </p>
          </div>
          <div
            className="margin-bottom--half"
          >
            <h4
              className="t-sans t-small t-bold no-margin"
            >
              Phone
            </h4>
            <p>
              9954449943
            </p>
          </div>
          <div
            className="margin-bottom--half"
          >
            <h4
              className="t-sans t-small t-bold no-margin"
            >
              Phone Type
            </h4>
            <p>
              Cell
            </p>
          </div>
        </div>
      </li>
      <li
        className="content-item"
      >
        <div
          className="content-card"
        >
          <div
            className="margin-bottom--half"
          >
            <h4
              className="t-sans t-small t-bold no-margin"
            >
              Agency Name
            </h4>
            <p>
              None
            </p>
          </div>
          <div
            className="margin-bottom--half"
          >
            <h4
              className="t-sans t-small t-bold no-margin"
            >
              Email
            </h4>
            <p>
              fede@eee.com
            </p>
          </div>
          <div
            className="margin-bottom--half"
          >
            <h4
              className="t-sans t-small t-bold no-margin"
            >
              Alternate Contact Type
            </h4>
            <p>
              Friend
            </p>
          </div>
          <div
            className="margin-bottom--half"
          >
            <h4
              className="t-sans t-small t-bold no-margin"
            >
              Alternate Contact Type Other
            </h4>
            <p>
              None
            </p>
          </div>
          <div
            className="margin-bottom--half"
          >
            <h4
              className="t-sans t-small t-bold no-margin"
            >
              Mailing Address
            </h4>
            <p>
              123 Alternate Mailing St, SF Mailing, CA, 94105-1804
            </p>
          </div>
        </div>
      </li>
    </ul>
  </div>
  <div
    className="content-card"
    id="content-card-household_members"
  >
    <h4
      className="content-card_title t-serif"
    >
      Household Members
    </h4>
    <table>
      <thead>
        <tr>
          <th>
            Name
          </th>
          <th>
            Relationship to Applicant
          </th>
          <th>
            Date of Birth
          </th>
          <th>
            Street
          </th>
          <th>
            City
          </th>
          <th>
            State
          </th>
          <th>
            Zip Code
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            diego m maradona
          </td>
          <td />
          <td>
            06/11/1976
          </td>
          <td>
            9 de julio
          </td>
          <td>
            buenos aires
          </td>
          <td>
            fl
          </td>
          <td>
            33026
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="content-card padding-bottom-none margin-bottom--half bg-trans"
  >
    <h4
      className="content-card_title t-serif"
    >
      Reserved and Priority Qualifying Information
    </h4>
    <ul
      className="content-grid"
    >
      <li
        className="content-item"
      >
        <div
          className="content-card"
        >
          <div
            className="margin-bottom--half"
          >
            <h4
              className="t-sans t-small t-bold no-margin"
            >
              Has Military Service
            </h4>
            <p>
              None
            </p>
          </div>
          <div
            className="margin-bottom--half"
          >
            <h4
              className="t-sans t-small t-bold no-margin"
            >
              Has Developmental Disability
            </h4>
            <p>
              None
            </p>
          </div>
        </div>
      </li>
      <li
        className="content-item"
      >
        <div
          className="content-card"
        >
          <div
            className="margin-bottom--half"
          >
            <h4
              className="t-sans t-small t-bold no-margin"
            >
              Has ADA Priorities Selected
            </h4>
            <p>
              Vision impairments;Mobility impairments;Hearing impairments
            </p>
          </div>
          <div
            className="margin-bottom--half"
          >
            <h4
              className="t-sans t-small t-bold no-margin"
            >
              Answered Community Screening
            </h4>
            <p>
              None
            </p>
          </div>
        </div>
      </li>
    </ul>
  </div>
  <div
    className="content-card"
    id="content-card-preferences"
  >
    <h4
      className="content-card_title t-serif"
    >
      Application Preferences
    </h4>
    <table>
      <thead>
        <tr>
          <th>
            Receives Preference
          </th>
          <th>
            Preference Name
          </th>
          <th>
            Lottery Status
          </th>
          <th>
            Person who claimed Name
          </th>
          <th>
            Type of proof
          </th>
          <th>
            Preference Lottery Rank
          </th>
          <th>
            Opt Out
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td />
          <td>
            Alice Griffith Housing Development Resident
          </td>
          <td>
            None
          </td>
          <td />
          <td />
          <td />
          <td />
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="content-card padding-bottom-none margin-bottom--half bg-trans"
  >
    <h4
      className="content-card_title t-serif"
    >
      Declared Household Income
    </h4>
    <ul
      className="content-grid"
    >
      <li
        className="content-item"
      >
        <div
          className="content-card"
        >
          <div
            className="margin-bottom--half"
          >
            <h4
              className="t-sans t-small t-bold no-margin"
            >
              Annual Income
            </h4>
            <p>
              110000
            </p>
          </div>
        </div>
      </li>
      <li
        className="content-item"
      >
        <div
          className="content-card"
        >
          <div
            className="margin-bottom--half"
          >
            <h4
              className="t-sans t-small t-bold no-margin"
            >
              Monthly Income
            </h4>
            <p>
              None
            </p>
          </div>
          <div
            className="margin-bottom--half"
          >
            <h4
              className="t-sans t-small t-bold no-margin"
            >
              Housing Voucher or Subsidy
            </h4>
            <p>
              true
            </p>
          </div>
        </div>
      </li>
    </ul>
  </div>
  <div
    className="content-card"
  >
    <h4
      className="content-card_title t-serif"
    >
      Attachments
    </h4>
    <ul>
      <li>
        <a
          href="undefined/servlet/servlet.FileDownload?file=xxxx1"
          target="_blank"
        >
          Namexxx1
        </a>
      </li>
      <li>
        <a
          href="undefined/sfc/servlet.shepherd/version/download/xxxx2"
          target="_blank"
        >
          Namexxx2
        </a>
      </li>
    </ul>
  </div>
</div>
`;
